packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -464,19 +465,78 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
         // TODO: This snapshot will return deltas, which we currently aren't using. We need to enable this flag to go down the "optimized"
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
-        const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
+        const { headers, url, isOptionsCall } = getUrlAndHeadersAndOptionsWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
 
         // This event measures only successful cases of getLatest call (no tokens, no retries).
         const { snapshot, canCache } = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "TreesLatest" }, async (event) => {
             const response = await fetchHelper<IOdspSnapshot>(url, { headers });
             const content = response.content;
+            let isResourceDataAvailable = false;
+            let dnstime; // domainLookupEnd - domainLookupStart
+            let redirectTime; // redirectEnd -redirectStart
+            let tcpHandshakeTime; // connectEnd  - connectStart
+            let secureConntime; // connectEnd  - secureConnectionStart
+            let responseTime; // responsEnd - responseStart
+            let fetchStToRespEndTime; // responseEnd  - fetchStart
+            let reqStToRespEndTime; // responseEnd - requestStart
+            let stToRespEnd; // responseEnd - startTime
+            const spReqDuration = response.headers.get("sprequestduration");
+
+            if (performance) {
+                const resources1 = performance.getEntriesByType("resource");
```

Comment:
It should be always there? Why are you checking this in if?

Reply:
I was not sure if it's always there, that's why I added the check. Based on your comment, I'm assuming it's always available, so removed the check now.